packages/gestalt/src/Callout.js

```diff
@@ -0,0 +1,159 @@
+// @flow strict
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Box from './Box.js';
+import GestaltProvider from './GestaltProvider';
+import Heading from './Heading.js';
+import Icon from './Icon.js';
+import IconButton from './IconButton.js';
+import Link from './Link.js';
+import Text from './Text.js';
+import { useColorScheme } from './contexts/ColorScheme.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
+
+type Props = {|
+  description: string,
+  dismissButton?: {|
+    accessibilityLabel: string,
+    onDismiss: () => void,
+  |},
+  iconAccessibilityLabel: string,
+  primaryLink?: {|
+    href: string,
+    label: string,
+    onClick?: AbstractEventHandler<
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>
+    >,
+  |},
+  secondaryLink?: {|
+    href: string,
+    label: string,
+    onClick?: AbstractEventHandler<
+      | SyntheticMouseEvent<HTMLAnchorElement>
+      | SyntheticKeyboardEvent<HTMLAnchorElement>
+    >,
+  |},
+  style: 'error' | 'info' | 'warning',
+  title?: string,
+|};
+
+const CALLOUT_TYPE_ATTRIBUTES = {
+  info: {
+    icon: 'check-circle',
+    color: 'blue',
+    backgroundColor: '#EBF4FE',
+  },
+  warning: {
+    icon: 'workflow-status-warning',
+    color: 'orange',
+    backgroundColor: '#FDF5EC',
+  },
+  error: {
+    icon: 'workflow-status-problem',
+    color: 'red',
+    backgroundColor: '#FDEBEE',
+  },
+};
+
+export default function Callout({
+  description,
+  dismissButton,
+  iconAccessibilityLabel,
+  primaryLink,
+  secondaryLink,
+  style,
+  title,
+}: Props): React.Node {
+  // Currently there is not a dark mode spec for this component. This is to ensure
+  // that all text is readable.
+  const { name } = useColorScheme();
+  const isDarkMode = name === 'darkMode';
+
+  return (
+    <Box
+      alignItems="center"
+      display="flex"
+      position="relative"
+      rounding={4}
+      dangerouslySetInlineStyle={{
+        __style: {
+          backgroundColor: CALLOUT_TYPE_ATTRIBUTES[style].backgroundColor,
+        },
+      }}
+      padding={8}
+    >
+      <Box>
+        <Icon
+          icon={CALLOUT_TYPE_ATTRIBUTES[style].icon}
+          color={CALLOUT_TYPE_ATTRIBUTES[style].color}
+          accessibilityLabel={iconAccessibilityLabel}
+          size={32}
+        />
+      </Box>
+      <Box flex="grow" paddingX={6} maxWidth={648}>
+        {title && (
+          <Box marginBottom={2}>
+            <Heading color={isDarkMode ? 'white' : 'darkGray'} size="sm">
+              {title}
+            </Heading>
+          </Box>
+        )}
+        <Text color={isDarkMode ? 'white' : 'darkGray'}>{description}</Text>
+      </Box>
+      {secondaryLink && (
+        <Box paddingX={1} marginStart="auto">
+          <Box padding={4} flex="none">
+            <Link href={secondaryLink.href}>
```

Comment:
@norrise120 could we add the `onClick` we passed through? `onClick={secondaryLink.onClick}`